crates/batch-prover/src/rpc.rs

```diff
@@ -196,6 +230,121 @@ where
         Ok(job_ids)
     }
 
+    async fn prove_native(&self, index_start: u32, index_end: u32) -> RpcResult<String> {
+        let ledger_db = &self.context.ledger_db;
+
+        if index_start > index_end {
+            return Err(internal_rpc_error("Invalid index range"));
+        }
+        // don't allow first commitment index to be called through this rpc as it requires extra handling
+        if index_start <= 1 {
+            return Err(internal_rpc_error(
+                "proveNative rpc supports only index_start > 1",
+            ));
+        }
+
+        let previous_commitment = ledger_db
+            .get_commitment_by_index(index_start - 1)
+            .map_err(|e| internal_rpc_error(e.to_string()))?
+            .ok_or_else(|| internal_rpc_error("Missing previous commitment index"))?;
+
+        let commitments = ledger_db
+            .get_commitment_by_range(index_start..=index_end)
+            .map_err(|e| internal_rpc_error(e.to_string()))?;
+        if commitments.len() as u32 != index_end - index_start + 1 {
+            return Err(internal_rpc_error(
+                "Missing some commitment indices from the range",
+            ));
+        }
+
+        let last_commitment = commitments.last().expect("Must have at least 1");
```

Comment:
```suggestion
        let last_commitment = commitments.last().expect("Must have at least 1");
```
Use ok_or and return an error instead of panicing here

Reply:
but with this:
```rs
if index_start > index_end {
       return Err(internal_rpc_error("Invalid index range"));
}
```
and this checks:
```rs
if commitments.len() as u32 != index_end - index_start + 1 {
             return Err(internal_rpc_error(
                 "Missing some commitment indices from the range",
             ));
}
```
it is already ensured that there is definitely at least one commitment.